packages/twenty-server/src/engine/metadata-modules/user-role/user-role.service.ts

```diff
@@ -96,33 +98,47 @@ export class UserRoleService {
     });
   }
 
-  public async getRolesForUserWorkspace(
-    userWorkspaceId: string,
-  ): Promise<RoleDTO[] | []> {
-    const userWorkspaceRole = await this.userWorkspaceRoleRepository.findOne({
+  public async getRolesByUserWorkspaces(
+    userWorkspaceIds: string[],
+  ): Promise<Map<string, RoleDTO[]>> {
+    const userWorkspaceRoles = await this.userWorkspaceRoleRepository.find({
       where: {
-        userWorkspaceId,
+        userWorkspaceId: In(userWorkspaceIds),
       },
     });
```

Comment:
```suggestion
      relations: {
        role: true,
      },
    });
```
Here we have the relation so you can add these lines to avoid the query below with roleRepository